test/builtin_dataset_mocks.py

```diff
@@ -1344,3 +1344,96 @@ def pcam(info, root, config):
             compressed_file.write(compressed_data)
 
     return num_images
+
+
+class CityScapesMockData:
+
+    _ARCHIVE_NAMES = {
+        ("Coarse", "train"): [("gtCoarse.zip", "gtCoarse"), ("leftImg8bit_trainvaltest.zip", "leftImg8bit")],
+        ("Coarse", "train_extra"): [("gtCoarse.zip", "gtCoarse"), ("leftImg8bit_trainextra.zip", "leftImg8bit")],
+        ("Coarse", "val"): [("gtCoarse.zip", "gtCoarse"), ("leftImg8bit_trainvaltest.zip", "leftImg8bit")],
+        ("Fine", "train"): [("gtFine_trainvaltest.zip", "gtFine"), ("leftImg8bit_trainvaltest.zip", "leftImg8bit")],
+        ("Fine", "test"): [("gtFine_trainvaltest.zip", "gtFine"), ("leftImg8bit_trainvaltest.zip", "leftImg8bit")],
+        ("Fine", "val"): [("gtFine_trainvaltest.zip", "gtFine"), ("leftImg8bit_trainvaltest.zip", "leftImg8bit")],
+    }
+
+    @classmethod
+    def generate(cls, root, config):
+
+        mode = config.mode.capitalize()
+        split = config.split
+
+        if split in ["train", "train_extra"]:
+            cities = ["bochum", "bremen"]
+            num_samples = 3
+        else:
+            cities = ["bochum"]
+            num_samples = 2
+
+        polygon_target = {
+            "imgHeight": 1024,
+            "imgWidth": 2048,
+            "objects": [
+                {
+                    "label": "sky",
+                    "polygon": [
+                        [1241, 0],
+                        [1234, 156],
+                        [1478, 197],
+                        [1611, 172],
+                        [1606, 0],
+                    ],
+                },
+                {
+                    "label": "road",
+                    "polygon": [
+                        [0, 448],
+                        [1331, 274],
+                        [1473, 265],
+                        [2047, 605],
+                        [2047, 1023],
+                        [0, 1023],
+                    ],
+                },
+            ],
+        }
+
+        gt_dir = root / f"gt{mode}"
+
+        for city in cities:
+
+            def make_image(name, size=10):
+                create_image_folder(
+                    root=gt_dir / split,
+                    name=city,
+                    file_name_fn=lambda idx: name.format(idx=idx),
+                    size=size,
+                    num_examples=num_samples,
+                )
+
+            make_image(f"{city}_000000_00000" + "{idx}" + f"_gt{mode}_instanceIds.png")
```

Comment:
Maybe use a lambda here to avoid piecing the string together?

```suggestion
            make_image(lambda idx: f"{city}_000000_{idx:06d}_gt{mode}_instanceIds.png")
```